Source/NSFileWrapper.m

```diff
@@ -0,0 +1,768 @@
+{
+  // FIXME
+  // For 
+  return NO;
+}
+
+- (BOOL) readFromURL: (NSURL*)url
+             options: (NSFileWrapperReadingOptions)options
+               error: (NSError**)outError
+{
+  // FIXME
+  NSFileManager *fm = [NSFileManager defaultManager];
+  NSString *path = [url path];
+  NSDictionary *fileAttributes;
+  NSString *fileType;
+
+  NSDebugLLog(@"NSFileWrapper", @"readFromURL: %@", path);
+
+  fileAttributes = [fm fileAttributesAtPath: path traverseLink: NO];
+  fileType = [fileAttributes fileType];
+  if ([fileType isEqualToString: NSFileTypeDirectory])
+    {
+      if (options & NSFileWrapperReadingImmediate)
+        {
+          NSString *filename;
+          NSMutableArray *fileWrappers = [NSMutableArray array];
+          NSArray *filenames = [fm directoryContentsAtPath: path];
+          NSEnumerator *enumerator = [filenames objectEnumerator];
+
+          while ((filename = [enumerator nextObject]) != nil)
+            {
+              NSFileWrapper *w;
+
+              w = [[NSFileWrapper alloc] initWithPath: 
+                                           [path stringByAppendingPathComponent: filename]];
+              [fileWrappers addObject: w];
+              RELEASE(w);
+            }
+          self = [self initDirectoryWithFileWrappers: 
+                         [NSDictionary dictionaryWithObjects: fileWrappers
+                                                     forKeys: filenames]];
+        }
+      else
+        {
+          self = [self initDirectoryWithFileWrappers: nil];
+        }
+    }
+  else if ([fileType isEqualToString: NSFileTypeRegular])
+    {
+      if (options & NSFileWrapperReadingWithoutMapping)
+        {
+          self = [self initRegularFileWithContents: 
+                         AUTORELEASE([[NSData alloc] initWithContentsOfFile: path])];
+        }
+      else
+        {
+          self = [self initRegularFileWithContents: 
+                         AUTORELEASE([[NSData alloc] initWithContentsOfMappedFile: path])];
+        }
+    }
+  else if ([fileType isEqualToString: NSFileTypeSymbolicLink])
+    {
+      self = [self initSymbolicLinkWithDestination: 
+                 [fm pathContentOfSymbolicLinkAtPath: path]];
+    }
+
+  // Store the full path in filename, the specification is unclear in this point
+  [self setFilename: path];
+  [self setPreferredFilename: [path lastPathComponent]];
+  [self setFileAttributes: [fm fileAttributesAtPath: path traverseLink: NO]];
+
+  return NO;
+}
+
+- (BOOL) writeToURL: (NSURL*)url
+            options: (NSFileWrapperWritingOptions)options
+originalContentsURL: (NSURL*)originalContentsURL
+              error: (NSError**)outError
+{
+  // FIXME
+  return [self writeToFile: [url path]
+                atomically: options & NSFileWrapperWritingAtomic
+           updateFilenames: options & NSFileWrapperWritingWithNameUpdating];
+}
+
+- (void) setIcon: (id)image
+{
+  // this method is here to quell compiler warnings.
```

Comment:
Now that you renamed the instance variable again, the old implementation of this method would work again.